desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -99,26 +101,35 @@ def handle_on_message(channel_id, bot_id, text, user_id):
       send_hi_user(channel_id, user_id)
 
 
-def handle_on_link_shared(channel_id, message_ts, links):
+def handle_on_link_shared(channel_id, message_ts, links, user_id):
   for item in links:
     path = urlsplit(item['url'])[2]
     id_type, qid = urlsplit(item['url'])[3].split('=')
+    query_id = {'id': qid} if qid.isdigit() else {'uuid': qid}
 
     try:
       if path == '/hue/editor' and id_type == 'editor':
-        doc = Document2.objects.get(id=qid)
+        doc = Document2.objects.get(**query_id)
         doc_type = 'Query'
       elif path == '/hue/gist' and id_type == 'uuid':
-        doc = _get_gist_document(uuid=qid)
+        doc = _get_gist_document(**query_id)
         doc_type = 'Gist'
       else:
         raise PopupException(_("Cannot unfurl link"))
     except Document2.DoesNotExist:
-      msg = "Document with {key}={value} does not exist".format(key='uuid' if id_type == 'uuid' else 'id', value=qid)
+      msg = "Document with {} does not exist".format(query_id)
```

Comment:
Empty?

%(key)s ...

When you are not sure, please try in a shell or better include it in the tests..